src/fft_cufft.f90

```diff
@@ -36,6 +36,14 @@ module decomp_2d_fft
    integer*4, save :: plan(-1:2, 3)
    complex*8, device, allocatable, dimension(:) :: cufft_workspace
 
+   ! Supported FFT backends
+   integer, parameter, public :: D2D_FFT_BACKEND_GENERIC = 0
+   integer, parameter, public :: D2D_FFT_BACKEND_FFTW3 = 1
+   integer, parameter, public :: D2D_FFT_BACKEND_FFTW3_F03 = 2
+   integer, parameter, public :: D2D_FFT_BACKEND_MKL = 3
+   integer, parameter, public :: D2D_FFT_BACKEND_CUFFT = 4
```

Comment:
I would define these once in a separate module, rather than keep redefining

Reply:
The current workaround to avoid repetitions would be to have them defined in decomp_2d. But I do not want to put decomp_2d_fft stuff inside decomp_2d...

Reply:
Or in a file and #include the file

Reply:
Otherwise, we need to add a file with all the public symbols / parameters in a dedicated module ?

Reply:
> Or in a file and #include the file

Definitely not :)

> Otherwise, we need to add a file with all the public symbols / parameters in a dedicated module ?

This is what I would suggest , e.g. `decomp_2d_constants` module. The decomp_2d module is too big, I don't think we should be exposing everything there and should eventually move things into the new module. However as this is a new variable there is no harm in putting it in the right place to begin with